test-jest/component/ResponsiveContainer.spec.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+import { ResponsiveContainer } from '../../src';
+
+declare global {
+  interface Window {
+    ResizeObserver: unknown;
+  }
+}
+
+describe('<ResponsiveContainer />', () => {
+  let notifyResizeObserverChange: (arg: unknown) => void;
+
+  const resizeObserverMock = jest.fn().mockImplementation(callback => {
+    notifyResizeObserverChange = callback;
+
+    return {
+      observe: jest.fn().mockImplementation(),
+      unobserve: jest.fn(),
+      disconnect: jest.fn(),
+    };
+  });
+  const consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation(() => undefined);
+
+  beforeAll(() => {
+    delete window.ResizeObserver;
```

Comment:
This is a very interesting test setup. Could you please add a comment as to the motivation and usage? 

Thus others might learn from this for further tests.